client/src/components/GameBoard.js

```diff
@@ -32,27 +35,79 @@ const useStyles = makeStyles({
 export default function GameBoard() {
     const classes = useStyles();
     const { email } = useUserState();
-    const { matchState } = useContext(MatchContext);
+    const { matchState, setMatchState } = useContext(MatchContext);
+    const socketRef = useRef();
+    const { matchId } = useParams();
+    const [selected, setSelected] = useState({});
     // TODO: integrate with backend.
     //function onCardClick() {
     //}
 
+    useEffect(() => {
+        socketRef.current = io.connect("/");
+        if (!matchState) {
+            socketRef.current.emit("get-game-engine", { matchId });
+        }
+        socketRef.current.on("update-game-engine-" + matchId, (game) => {
+            setMatchState(game);
+            if (matchState.votes === {}) {
+                setSelected({});
+            }
+        });
+        return () => socketRef.current.disconnect();
+    }, [matchState, setMatchState, matchId]);
+
+    const handleVote = (word, row, column) => {
+        console.log(word, row, column);
+        setSelected({ row, column });
+        socketRef.current.emit("set-vote", { matchId, word, row, column, email });
+        return "x";
```

Comment:
why are we returning `x`?

Reply:
Was using it to test something before but forgot to remove it too